python/src/scipp/plot/plot_sparse.py

```diff
@@ -26,64 +30,90 @@ def plot_sparse(input_data, ndim=0, sparse_dim=None, backend=None, logx=False,
     coords = var.coords
 
     # Parse colorbar
-    cbar = parse_colorbar(cb, plotly=True)
+    cbar = parse_colorbar(cb, var, values=sparse_data[-1])
+
+    members = {}
+    ipv = None
 
-    xyz = "xyz"
     if ndims < 3:
-        plot_type = "scattergl"
+        fig, ax = plt.subplots(1, 1, figsize=(config.width/config.dpi,
+                                              config.height/config.dpi),
+                               dpi=config.dpi)
+        widg = None
+        members["ax"] = ax
+        params = dict(label=name, edgecolors="#ffffff", c=color)
+        xs = sparse_data[ndims - 1]
+        ys = sparse_data[ndims - 2]
+        if len(sparse_data) > ndims:
+            if weights.count("size") > 0:
+                params["s"] = sparse_data[-1] * size
+                params["alpha"] = opacity
+            if weights.count("color") > 0:
+                params["c"] = sparse_data[-1]
+                params["cmap"] = cbar["name"]
+                params["norm"] = cbar["norm"]["values"]
+
+        if marker is None:
+            marker = "o"
+        scat = ax.scatter(xs, ys, **params)
+        if len(sparse_data) > ndims and weights.count("color") > 0:
+            colorbar = plt.colorbar(scat, ax=ax, cax=None)
+            colorbar.ax.set_ylabel(axis_label(name="Weights", log=cbar["log"]))
+            members["colorbars"] = colorbar
+
+        ax.set_xlabel(axis_label(coords[sparse_dim]))
+        if ndims > 1:
+            ax.set_ylabel(axis_label(coords[dims[int(ndims == 3)]]))
+
+        ax.legend()
+        if title is not None:
+            ax.set_title(title)
+        if logx or logxy:
+            ax.set_xscale("log")
+        if logy or logxy:
+            ax.set_yscale("log")
+
     elif ndims == 3:
-        plot_type = "scatter3d"
+
+        import ipyvolume as ipv
+
+        fig = ipv.figure(width=config.width, height=config.height,
+                         animation=0)
+
+        params = dict()
+        if isinstance(color, list):
+            params["color"] = color[0]
+        else:
+            params["color"] = color
+        params["x"] = sparse_data[ndims - 1]
+        params["y"] = sparse_data[ndims - 2]
+        params["z"] = sparse_data[0]
+        if len(sparse_data) > ndims and weights:
+            scalar_map = cm.ScalarMappable(norm=cbar["norm"]["values"],
+                                           cmap=cbar["name"])
+            params["color"] = scalar_map.to_rgba(sparse_data[-1])
+
+        if marker is None:
+            marker = "sphere"
+        params["marker"] = marker
+        scat = ipv.scatter(**params)
```

Comment:
(unrelated) I was wondering if we could use scatter plots to make an "InstrumentView"?

Reply:
yes, i had tried something like that in the past, and it should definitely be possible.